packages/core-data/src/resolvers.js

```diff
@@ -585,7 +585,16 @@ export const getNavigationFallbackId =
 		dispatch.receiveNavigationFallbackId( fallback?.id );
 
 		if ( record ) {
-			const invalidateNavigationQueries = true;
+			// If the fallback is already in the store, don't invalidate navigation queries.
+			// Otherwise, invalidate the cache for the scenario where there were no Navigation
+			// posts in the state and the fallback created one.
+			const invalidateNavigationQueries = select.getEntityRecord(
+				'postType',
+				'wp_navigation',
+				fallback?.id
+			)
+				? false
+				: true;
```

Comment:
We can probably omit this `false` / `true` by inverting the result of `getEntityRecord`.